domain-management/src/main/java/org/jboss/as/domain/management/security/SecurityRealmChildAddHandler.java

```diff
@@ -56,6 +57,11 @@ public SecurityRealmChildAddHandler( boolean validateAuthentication, boolean val
     }
 
     @Override
+    protected boolean isResourceServiceRestartAllowed(OperationContext context, ServiceController<?> service) {
+        return false;
+    }
```

Comment:
@darranl WDYT about this:

I don't think this should override the SecurityRealmParentRestartHandler impl. Instead, lets change ManagementUtil.isSecurityRealmReloadRequired so it ignores context.isResourceServiceRestartAllowed and returns true if the service is present and UP. Then in SecurityRealmAddHandler L219 we change it to

```
realmBuilder.setInitialMode(Mode.ON_DEMAND);
```

PRO: 
1) People can still build up a realm without needing a composite or a reload, since until it is demanded, child add ops will reinstall the service. So this may avoid breaking some scripts.
2) We don't start undemanded services.

CON: The realm service isn't actively started so we don't aggressively detect problems with it. They are discovered later when the admin adds something that demands it.